pgrx-tests/src/framework.rs

```diff
@@ -381,19 +387,59 @@ fn initdb(postgresql_conf: Vec<&'static str>) -> eyre::Result<()> {
     let pgdata = get_pgdata_path()?;
 
     if !pgdata.is_dir() {
+        let base = pgdata.parent().unwrap();
+        if !base.exists() {
+            std::fs::create_dir(base)?;
+        }
+
+        if let Some(runas) = get_runas() {
+            // we've been asked to run as a different user.  As such, the PGDATA directory we just created
+            // needs to be owned by that user.
+            //
+            // In order to do that, we have to become "root" to change its ownership.
+            let mut chown = Command::new("sudo");
+            chown
+                .args(&["-u", "root", "chown", &runas])
```

Comment:
Pardon if I am being thick, but what exactly prevents this from being just
```suggestion
            let mut chown = Command::new("chown");
            chown
                .arg(&runas)
```

Reply:
You gotta be root in order to change the ownership of a file to a different user.

Reply:
Hm. For some reason I thought you could chown files you own to another user. Guess not.